src/app/game/game.component.ts

```diff
@@ -26,25 +52,164 @@ export class GameComponent implements OnInit {
    * @param route - allows to verify player by id using the url which opened the game
    */
   constructor(
+    public gameService: GameService,
+    public translate: TranslateService,
     private route: ActivatedRoute,
+    private router: Router,
     private playerService: PlayerService,
-    public messageService: MessageService,
-    public translate: TranslateService) {
+    private notificationService: NotificationService) {
+      this.hitMessage = 'TOAST.HIT';
+      this.missMessage = 'TOAST.MISS';
+      this.turnMessage = 'TOAST.YOUR_TURN';
+      this.enemyTurnMessage = 'TOAST.ENEMY_TURN';
    }
 
   /**
    * Calls for getPlayer() method on component initialization
    */
   ngOnInit(): void {
-    this.getPlayer();
+    this.player = {name : this.route.snapshot.paramMap.get('name')};
+    this.getOpponent();
+    this.initializeEmptyMaps();
+    this.getShootMap();
+    this.getShipMap();
+
+    this.subscription = timer(0, 2000)
+    .pipe(switchMap(() => this.gameService.getCurrentGameStatus()))
+    .subscribe( currentGameStatus => {
+        if(currentGameStatus.playerNameWhoMoves === this.player.name && !currentGameStatus.playerLooser){
+          if(!this.isGameSetToPLayersTurn)
+          {
+            this.showMessage(this.turnMessage, NotificationType.info);
+            this.getShipMap();
+            this.unlockAllShotSquares();
+            this.isGameSetToPLayersTurn = true;
+            this.isFirstTurn = false;
+          }
+        }
+        else if(currentGameStatus.playerNameWhoMoves === this.opponent.name && !currentGameStatus.playerLooser){
+          this.blockAllShotSquares();
+          this.isGameSetToPLayersTurn = false;
+          if(this.isFirstTurn){
+            this.showMessage(this.enemyTurnMessage, NotificationType.info);
+            this.isFirstTurn = false;
+          }
+        }
+        else if(currentGameStatus.playerNameWhoMoves === this.opponent.name && currentGameStatus.playerLooser){
+          this.router.navigate(['/landing/win']);
+        }
+        else if(currentGameStatus.playerNameWhoMoves === this.player.name && currentGameStatus.playerLooser){
+          this.router.navigate(['/landing/loose']);
+        }
+        this.playerTurn = {name : currentGameStatus.playerNameWhoMoves};
+    } );
+  }
+
+  ngOnDestroy(): void{
```

Comment:
After the player surrenders, it is removed from the db. When you try to access his game screen it again redirects to landing/loose.
However, the winner is redirected at first, but it is possible to access his game screen after that.